test/functional/collations_tests.js

```diff
@@ -1102,6 +1103,33 @@ describe('Collation', function() {
     }
   });
 
+  it('cursor count method should return the correct number when used with collation set', {
+    metadata: { requires: { mongodb: '>=3.4.0' } },
+    test: function(done) {
+      const configuration = this.configuration;
+      const client = configuration.newClient({ w: 1 }, { poolSize: 1, auto_reconnect: false });
+
+      client.connect(function(err, client) {
+        const db = client.db(configuration.db);
+        const docs = [{ _id: 0, name: 'foo' }, { _id: 1, name: 'Foo' }];
+        const collation = { locale: 'en_US', strength: 2 };
+        const close = e => cursor.close(() => client.close(() => done(e)));
```

Comment:
Let's switch this line with the one beneath it, so that it goes:

```js
let collection, cursor;
const close = e => cursor.close(() => client.close(() => done(e)));
```